[18.0][IMP] account_avatax_demo_oca: Update demo data and unit tests

natuan9 opened the pull request.

atchuthan committed:
Add Avatax Exemption module to 14.0

atchuthan committed:
Add Avatax Exemption module to 14.0

atchuthan committed:
String parameter on UserError requires translation.

atchuthan committed:
modified exemption fields

atchuthan committed:
0.1.10 version change

atchuthan committed:
0.1.10 version change

atchuthan committed:
0.1.11 version update

atchuthan committed:
0.1.12 update

atchuthan committed:
0.1.12 update: corrections

atchuthan committed:
2.0.0 update

atchuthan committed:
2.0.1 update

atchuthan committed:
remove readonly for taxable field

amkarthik committed:
AttributeError: 'CustomerPortal (extended by CustomerPortal, Portal' object has no attribute '_get_archive_groups'

cybernexus committed:
account_avatax_website_sale: module to recalculate taxes prior to taking payment for website order

atchuthan committed:
readme.rst

atchuthan committed:
readme.rst

atchuthan committed:
Moved odoo enterprise related fields to custom module

atchuthan committed:
Moved odoo enterprise related fields to custom module

oca-travis committed:
Update account_avatax_website_sale.pot

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
icon.png

oca-travis committed:
Update account_avatax_exemption_base.pot

oca-travis committed:
Update account_avatax_exemption.pot

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
README.rst

ValentinVinagre committed:
Account Fiscal Position Partner Type

dreispt committed:
account_avatax_exemption_base: license must be AGPL because of dependencies

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
account_avatax_exemption_base 14.0.1.0.1

oca-travis committed:
Update account_fiscal_position_partner_type.pot

OCA-git-bot committed:
README.rst

pedrobaeza committed:
account_fiscal_position_partner_type: Proper version

vimalpatelserpentcs committed:
account_fiscal_position_partner_type: module migrated into v14

vimalpatelserpentcs committed:
account_fiscal_position_partner_type: improved the code as per comment

oca-travis committed:
Update account_fiscal_position_partner_type.pot

OCA-git-bot committed:
README.rst

atchuthan committed:
account_avatax_exemption: Allow cancel option in Inprogress stage

atchuthan committed:
account_avatax_exemption: Allow cancel option in Inprogress stage

dreispt committed:
Added translation using Weblate (Portuguese (Portugal))

dreispt committed:
Translated using Weblate (Portuguese (Portugal))

Currently translated at 100.0% (11 of 11 strings)

Translation: account-fiscal-rule-14.0/account-fiscal-rule-14.0-account_fiscal_position_partner_type
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-14-0/account-fiscal-rule-14-0-account_fiscal_position_partner_type/pt_PT/

atchuthan committed:
Avatax: Use Odoo Tax option

OCA-git-bot committed:
account_avatax_exemption_base 14.0.1.1.0

OCA-git-bot committed:
account_avatax_exemption 14.0.1.1.0

manuelregidor committed:
account_fiscal_position_partner_type: Migration to 15.0

Commit:
Update account_fiscal_position_partner_type.pot

OCA-git-bot committed:
README.rst

atchuthan committed:
account_avatax_exemption: new parameter added

OCA-git-bot committed:
account_avatax_exemption 14.0.1.1.1

atchuthan committed:
account_avatax_exemption_base: change license to LGPL

atchuthan committed:
account_avatax_exemption_base: change license to LGPL

oca-travis committed:
Update account_avatax_exemption.pot

oca-travis committed:
Update account_avatax_exemption_base.pot

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
account_avatax_exemption 14.0.2.0.0

dreispt committed:
account_avatax_sale_oca: renamed from account_sale_avatax

dreispt committed:
account_avatax_exemption: account_avatax renamed to account_avatax_oca

Commit:
Update account_avatax_sale_oca.pot

dreispt committed:
account_avatax_website_sale: account_avatax renamed to account_avatax_oca

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
account_avatax_sale_oca 15.0.2.0.0

OCA-git-bot committed:
account_avatax_website_sale 14.0.2.0.0

OCA-git-bot committed:
account_avatax_exemption 14.0.3.0.0

RodrigoBM committed:
account_fiscal_position_partner_type: Migration to 16.0

Nikul-OSI committed:
account_avatax_sale_oca: black, isort, prettier

Nikul-OSI committed:
account_avatax_sale_oca: Migrated to v16

Commit:
Update account_fiscal_position_partner_type.pot

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
account_fiscal_position_partner_type 16.0.1.0.1

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-fiscal-rule-16.0/account-fiscal-rule-16.0-account_fiscal_position_partner_type
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-16-0/account-fiscal-rule-16-0-account_fiscal_position_partner_type/

sanrav committed:
Added translation using Weblate (French (France) (fr_FR))

sanrav committed:
Translated using Weblate (French (France) (fr_FR))

Currently translated at 100.0% (8 of 8 strings)

Translation: account-fiscal-rule-16.0/account-fiscal-rule-16.0-account_fiscal_position_partner_type
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-16-0/account-fiscal-rule-16-0-account_fiscal_position_partner_type/fr_FR/

ChrisOForgeFlow committed:
account_avatax_website_sale

Commit:
Update account_avatax_sale_oca.pot

OCA-git-bot committed:
README.rst

RLeeGML committed:
account_avatax_sale_oca: tax calculation improvements

* [FIX] compound rate computation adjustment

Commit:
Update account_avatax_website_sale.pot

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
README.rst

OCA-git-bot committed:
README.rst

AlexPForgeFlow committed:
account_avatax_exemption_base: Migration to 16.0

JordiBForgeFlow committed:
[ADD] account_avatax_oca_log

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-fiscal-rule-16.0/account-fiscal-rule-16.0-account_fiscal_position_partner_type
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-16-0/account-fiscal-rule-16-0-account_fiscal_position_partner_type/

OCA-git-bot committed:
post-merge updates

patrickrwilson committed:
account_avatax_sale_oca (Fixes Sale Line Access Error)

dreispt committed:
account_avatax_exemption: pre-commit fixes

Commit:
Update account_avatax_exemption.pot

OCA-git-bot committed:
post-merge updates

JordiBForgeFlow committed:
account_avatax_exemption_base: remove onchange method that must be moved to account_avatax_exemption

OCA-git-bot committed:
post-merge updates

metaminux committed:
account_avatax_sale_oca : override the create method in batch

OCA-git-bot committed:
post-merge updates

Commit:
Update account_avatax_exemption_base.pot

OCA-git-bot committed:
post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

Nikul-OSI committed:
[FIX] account_avatax_sale_oca: black, isort, prettier

Nikul-OSI committed:
[MIG] account_avatax_sale_oca: Migrated in v17

Nikul-OSI committed:
[FIX] account_fiscal_position_partner_type: black, isort, prettier

Nikul-OSI committed:
[FIX] account_avatax_website_sale: black, isort, prettier

Nikul-OSI committed:
[MIG] account_avatax_website_sale: Migrated to v17

Nikul-OSI committed:
[MIG] account_fiscal_position_partner_type: Migrated to v17

Nikul-OSI committed:
[FIX] account_avatax_exemption_base: black, isort, prettier

Nikul-OSI committed:
[MIG] account_avatax_exemption_base: Migrated in v17

Nikul-OSI committed:
[FIX] account_avatax_exemption: black, isort, prettier

Nikul-OSI committed:
[MIG] account_avatax_exemption: Migrated to v17

Commit:
[UPD] Update account_fiscal_position_partner_type.pot

OCA-git-bot committed:
[BOT] post-merge updates

dreispt committed:
[FIX] account_avatax_sale_oca: property_exemption_country_wide moved to account_avatax_oca
t

Commit:
[UPD] Update account_avatax_sale_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update account_avatax_website_sale.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-fiscal-rule-17.0/account-fiscal-rule-17.0-account_avatax_sale_oca
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-17-0/account-fiscal-rule-17-0-account_avatax_sale_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: account-fiscal-rule-17.0/account-fiscal-rule-17.0-account_fiscal_position_partner_type
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-17-0/account-fiscal-rule-17-0-account_fiscal_position_partner_type/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (29 of 29 strings)

Translation: account-fiscal-rule-17.0/account-fiscal-rule-17.0-account_avatax_sale_oca
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-17-0/account-fiscal-rule-17-0-account_avatax_sale_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (29 of 29 strings)

Translation: account-fiscal-rule-17.0/account-fiscal-rule-17.0-account_avatax_sale_oca
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-17-0/account-fiscal-rule-17-0-account_avatax_sale_oca/it/

dreispt committed:
[FX] account_avatax_sale_oca: Exemption is editable for Accounting Manager groupI

OCA-git-bot committed:
[BOT] post-merge updates

ChrisOForgeFlow committed:
[IMP] added mockup of avatax anwser

AlexPForgeFlow committed:
[FIX] minor fix

kobros-tech committed:
[17.0][MIG] account_avatax_oca_log: Migration to 17.0

kobros-tech committed:
[18.0][MIG] account_avatax_oca_log: Migration to 18.0

kobros-tech committed:
[18.0][MIG] account_avatax_sale_oca: Migration to 18.0

kobros-tech committed:
[18.0][MIG] : account_avatax_website_sale Migration to 18.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (190 of 190 strings)

Translation: account-fiscal-rule-18.0/account-fiscal-rule-18.0-account_avatax_oca
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-18-0/account-fiscal-rule-18-0-account_avatax_oca/it/

dnplkndll committed:
[IMP] account_avatax_exemption: lint

fix: lint

dnplkndll committed:
[IMP] account_avatax_exemption_base: pre-commit auto fixes

dnplkndll committed:
[MIG] account_avatax_exemption: Migration to 18.0

fix: model_create_multi

fix: _search()

dnplkndll committed:
[MIG] account_avatax_exemption_base: Migration to 18.0

fix: perm groups missing on portal

[FIX] account_avatax_exemption_base: fix form view design

[IMP] code updates

[IMP] account_avatax_exemption_base: remove chatter from wizard view

dnplkndll committed:
[FIX] account_avatax_exemption_base lint

ThiagoMForgeFlow committed:
[MIG] account_fiscal_position_partner_type: Migration to 18.0

kobros-tech committed:
[18.0][Add] account_avatax_demo_oca: avatax demo module

OCA-git-bot committed:
[IMP] update dotfiles

dnplkndll committed:
[IMP] install readme

OCA-git-bot committed:
Merge PR #485 into 18.0

Signed-off-by dreispt

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #468 into 18.0

Signed-off-by dreispt

Commit:
[UPD] Update account_avatax_sale_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-fiscal-rule-18.0/account-fiscal-rule-18.0-account_avatax_sale_oca
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-18-0/account-fiscal-rule-18-0-account_avatax_sale_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (29 of 29 strings)

Translation: account-fiscal-rule-18.0/account-fiscal-rule-18.0-account_avatax_sale_oca
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-18-0/account-fiscal-rule-18-0-account_avatax_sale_oca/it/

kobros-tech committed:
[FIX] account_avatax_oca: return correct message format

OCA-git-bot committed:
Merge PR #488 into 18.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #494 into 18.0

Signed-off-by JordiBForgeFlow

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #476 into 18.0

Signed-off-by dreispt

Commit:
[UPD] Update account_fiscal_position_partner_type.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #460 into 18.0

Signed-off-by dreispt

Commit:
[UPD] Update account_avatax_oca_log.pot

OCA-git-bot committed:
[BOT] post-merge updates

kobros-tech committed:
[FIX] account_avatax_exemption_base: close curly brace

OCA-git-bot committed:
Merge PR #473 into 18.0

Signed-off-by dreispt

Commit:
[UPD] Update account_avatax_exemption_base.pot

OCA-git-bot committed:
[BOT] post-merge updates

natuan9 committed:
[IMP] account_avatax_demo_oca: Update demo data and unit tests

nguyenminhchien commented on test-requirements.txt:
```diff
@@ -0,0 +1,4 @@
+odoo-addon-account_avatax_exemption @ git+https://github.com/OCA/account-fiscal-rule.git@refs/pull/475/head#subdirectory=account_avatax_exemption
+odoo-addon-account_avatax_exemption_base @ git+https://github.com/OCA/account-fiscal-rule.git@refs/pull/473/head#subdirectory=account_avatax_exemption_base
```
this module has been merged to oca:18.0

nguyenminhchien commented on test-requirements.txt:
```diff
@@ -0,0 +1,4 @@
+odoo-addon-account_avatax_exemption @ git+https://github.com/OCA/account-fiscal-rule.git@refs/pull/475/head#subdirectory=account_avatax_exemption
+odoo-addon-account_avatax_exemption_base @ git+https://github.com/OCA/account-fiscal-rule.git@refs/pull/473/head#subdirectory=account_avatax_exemption_base
+odoo-addon-account_avatax_oca_log @ git+https://github.com/OCA/account-fiscal-rule.git@refs/pull/460/head#subdirectory=account_avatax_oca_log
```
this one was merged too

OCA-git-bot committed:
Merge PR #475 into 18.0

Signed-off-by dreispt

Commit:
[UPD] Update account_avatax_oca.pot

Commit:
[UPD] Update account_avatax_exemption.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-fiscal-rule-18.0/account-fiscal-rule-18.0-account_avatax_oca
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-18-0/account-fiscal-rule-18-0-account_avatax_oca/

nguyenminhchien commented on account_avatax_demo_oca/tests/SaleOrder_request.json:
```diff
@@ -0,0 +1,52 @@
+{
```
I found no place using this file

nguyenminhchien replied:
remove it if no need it

nguyenminhchien commented on account_avatax_demo_oca/tests/test_sale_order.py:
```diff
@@ -0,0 +1,138 @@
+            if line["description"] == "Taxed Product":
+                line["lineNumber"] = self.order.order_line[1].id
+
+        with self._capture_create_or_adjust_transaction(return_value=so_2_response):
```
```suggestion
        with self._capture_create_or_adjust_transaction(return_value=so_2_response) as captured:
```
capture the mock response to check the return values

nguyenminhchien replied:
test both cases: 
- Case 1: with self._capture_create_or_adjust_transaction(return_value=so_2_response) as captured
- Case 2: with self._capture_create_or_adjust_transaction(return_value=so_2_response, apply_args=True) as captured

Each case, test the values of:
- captured.mock_response.json()["lines"][...]["lineNumber"]
- captured.mock_response.json()["lines"][...]["itemCode"]
- captured.mock_response.json()["type"]
- captured.mock_response.json()["customerCode"]

nguyenminhchien commented on account_avatax_demo_oca/tests/test_sale_order.py:
```diff
@@ -0,0 +1,138 @@
+                line["lineNumber"] = self.order.order_line[1].id
+
+        with self._capture_create_or_adjust_transaction(return_value=so_2_response):
+            self.order._avatax_compute_tax()
```
Add tests to know tax records are created for sale order lines:
```
# test tax records are created for sale order lines
self.assertEqual(captured.mock_response.json()["lines"][0]["lineNumber"], self.order.order_line[0].id)
self.assertEqual(captured.mock_response.json()["lines"][1]["lineNumber"], self.order.order_line[1].id)
```

nguyenminhchien commented on account_avatax_demo_oca/tests/test_sale_order.py:
```diff
@@ -0,0 +1,138 @@
+        with self._capture_create_or_adjust_transaction(return_value=so_2_response):
+            self.order._avatax_compute_tax()
+            for order_line in self.order.order_line:
+                if order_line.name == "NT Product":
```
test tax_amt = 0 for the line of "NT Product"

nguyenminhchien commented on account_avatax_demo_oca/tests/test_sale_order.py:
```diff
@@ -0,0 +1,138 @@
+                        "AVATAX", tax_names, "AVATAX tax not found in Non-Tax Product"
+                    )
+
+                if order_line.name == "Taxed Product":
```
test tax_amt = 3.06 = so_2_response["totalTax"] for this line

nguyenminhchien commented on account_avatax_demo_oca/tests/test_sale_order.py:
```diff
@@ -0,0 +1,138 @@
+                    self.assertTrue(order_line.tax_id)
+                    self.assertNotIn(
+                        "AVATAX", tax_names, "AVATAX tax found in Taxed Product"
+                    )
```
test self.order.tax_amount = 3.06 = so_2_response["totalTax"]

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (191 of 191 strings)

Translation: account-fiscal-rule-18.0/account-fiscal-rule-18.0-account_avatax_oca
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-18-0/account-fiscal-rule-18-0-account_avatax_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (83 of 83 strings)

Translation: account-fiscal-rule-18.0/account-fiscal-rule-18.0-account_avatax_exemption_base
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-18-0/account-fiscal-rule-18-0-account_avatax_exemption_base/it/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (26 of 26 strings)

Translation: account-fiscal-rule-18.0/account-fiscal-rule-18.0-account_avatax_oca_log
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-18-0/account-fiscal-rule-18-0-account_avatax_oca_log/it/

nguyenminhchien reviewed (approved):
- I'm not sure about the groups of state.
- Others: lgtm

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (89 of 89 strings)

Translation: account-fiscal-rule-18.0/account-fiscal-rule-18.0-account_avatax_exemption
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-18-0/account-fiscal-rule-18-0-account_avatax_exemption/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (26 of 26 strings)

Translation: account-fiscal-rule-18.0/account-fiscal-rule-18.0-account_avatax_oca_log
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-18-0/account-fiscal-rule-18-0-account_avatax_oca_log/it/

OCA-git-bot committed:
Merge PR #467 into 18.0

Signed-off-by dreispt

Commit:
[UPD] Update account_avatax_exemption_base.pot

Commit:
[UPD] Update account_avatax_website_sale.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-fiscal-rule-18.0/account-fiscal-rule-18.0-account_avatax_exemption_base
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-18-0/account-fiscal-rule-18-0-account_avatax_exemption_base/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (83 of 83 strings)

Translation: account-fiscal-rule-18.0/account-fiscal-rule-18.0-account_avatax_exemption_base
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-18-0/account-fiscal-rule-18-0-account_avatax_exemption_base/it/
